src/ja/mangatoshokanz/src/eu/kanade/tachiyomi/extension/ja/mangatoshokanz/MangaToshokanZ.kt

```diff
@@ -0,0 +1,339 @@
+package eu.kanade.tachiyomi.extension.ja.mangatoshokanz
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.FormBody
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Element
+import java.lang.StringBuilder
+import java.security.KeyPair
+
+class MangaToshokanZ : HttpSource() {
+    override val lang = "ja"
+    override val supportsLatest = true
+    override val name = "マンガ図書館Z"
+    override val baseUrl = "https://www.mangaz.com"
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addNetworkInterceptor(::r18Interceptor)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        // author/illustrator name might just show blank if language not set to japan
+        .add("cookie", "_LANG_=ja")
+
+    private val keys: KeyPair by lazy {
+        getKeys()
+    }
+
+    private val _serial by lazy {
+        getSerial()
+    }
+
+    private var isR18 = false
+
+    private fun r18Interceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        // open access to R18 section
```

Comment:
shouldn't it be done before getting the response?

Reply:
ahh right, looks like i forgot to change it